contracts/DegenLockToken.sol

```diff
@@ -0,0 +1,194 @@
+     *  @dev The lock duration is longer than a year
+     */
+    error LockDurationIsTooLong();
+
+    /**
+     *  @dev Amount of tokens to be locked is zero
+     */
+    error ZeroAmount();
+
+    /**
+    /**
+     * @dev Minimum deposit not met
+     */
+    error MinimumDepositNotMet();
+
+    /**
+     *  @dev The lock duration has been updated
+     */
+    event LockDurationUpdated(uint256 indexed duration);
+
+    /**
+     *  @dev The minimum deposit amount has been updated
+     */
+    event MinDepositAmountUpdated(uint256 indexed minDepositAmount);
+
+    /**
+     *  @dev The deposit timestamp has been set
+     */
+    event DepositTimestampUpdated(
+        address indexed account,
+        uint256 indexed depositTimestamp
+    );
+
+    /**
+     * @dev Construct a new Degen token
+     */
+    constructor() ERC20(TOKEN_NAME, TOKEN_SYMBOL) Ownable(msg.sender) {
+        minDepositAmount = 10000 * 10 ** decimals();
+        emit MinDepositAmountUpdated(minDepositAmount);
+    }
+
+    /**
+     * @dev Deposit tokens to be locked until the end of the locking period
+     * @param amount The amount of tokens to deposit
+     */
+    function deposit(uint256 amount) external nonReentrant {
+        if (amount < minDepositAmount) {
+            revert MinimumDepositNotMet();
+        }
+
+        IERC20(TOKEN).safeTransferFrom(msg.sender, address(this), amount);
+        _mint(msg.sender, amount);
+
+        _depositTimestamps[msg.sender] = block.timestamp;
+        emit DepositTimestampUpdated(msg.sender, block.timestamp);
+    }
+
+    /**
+     * @dev Withdraw tokens after the end of the locking period or during the deposit period
+     * @param amount The amount of tokens to withdraw
+     */
+    function withdraw(uint256 amount) external nonReentrant {
+        if (amount == 0) {
+            revert ZeroAmount();
+        }
+
+        if (block.timestamp < _depositTimestamps[msg.sender] + lockDuration) {
+            revert LockPeriodOngoing();
+        }
+
+        _burn(msg.sender, amount);
+        IERC20(TOKEN).safeTransfer(msg.sender, amount);
+    }
+
+    /**
+     * @dev Update the lock duration
+     * @param newDuration The new lock duration in seconds
+     */
+    function updateLockDuration(uint256 newDuration) external onlyOwner {
+        if (newDuration <= 365 days) {
+            revert LockDurationIsTooLong();
+        }
+
+        lockDuration = newDuration;
+        emit LockDurationUpdated(lockDuration);
+    }
+
+    /**
+     * @dev Update the minimum deposit amount
+     * @param newMinDepositAmount The new minimum deposit amount
+     */
+    function updateMinDepositAmount(
+        uint256 newMinDepositAmount
+    ) external onlyOwner {
+        minDepositAmount = newMinDepositAmount;
+        emit MinDepositAmountUpdated(minDepositAmount);
+    }
+
+    /**
+     * @dev Last deposit timestamp for the account
+     */
+    function depositTimestamp(address account) external view returns (uint256) {
```

Comment:
You can remove this method by removing the leading underscore from `_depositTimestamps` and making it `public`.